src/pages/Game.tsx

```diff
@@ -42,28 +45,38 @@ function Game() {
     navigator.mediaDevices
       .getUserMedia({ video: { width: 300 }, audio: true })
       .then((stream) => {
-        videoRef.current.srcObject = stream;
-        videoRef.current.autoplay = true;
-        videoRef.current.muted = false;
-        setmystream(stream);
+        if (videoRef.current) {
+          videoRef.current.srcObject = stream;
+          videoRef.current.autoplay = true;
+          videoRef.current.muted = false;
+          setmystream(stream);
+        }
       })
       .catch((err) => {
-        console.log(err.message);
+        setMessage(err.message);
         setDisplayVid(false);
         setAudio(false);
       });
   }, []);
+  useEffect(() => {
+    socket.emit('join_room', {
+      room: id,
+    });
+    return () => {
+      socket.emit('leave_room', id);
+    };
+  }, [id]);
   function toggleVideo() {
     if (displayVid) {
       setDisplayVid(false);
-      mystream.getTracks().forEach((track: any) => {
+      mystream?.getTracks().forEach((track: MediaStreamTrack) => {
         if (track.readyState === 'live' && track.kind === 'video') {
           track.enabled = false;
         }
```

Comment:
And again, after scrapping the `if` and `else` blocks, something like this?
```suggestion
        track.enabled = !(track.readyState === 'live' && track.kind === 'video')
```
or similar?